### ISSUE TITLE
Question for clarification -- Regressing total counts and mitochondrial contribution

### ISSUE THREAD

**Issue by amjass12** [immunogenomics/harmony#92]
Hi all!

I have a quick question just for some clarification that I hope is ok posted here!

The harmony batch correction works really well with the Seurat workflow. Upstream of harmony when I scale data, I regress out total_counts as well as mitochondrial genes. In all vignettes that I see generally, scaleData is is just normally done as follows:

`scaleData('object')`

and i rarely see nUMI etc being regressed out. My question is, is it ok to do this prior to harmony batch correction if the only variable that you pass ot harmony is the 'batch' effect?

the pipeline i have been using is as follows:

```
merge<-NormalizeData(merge)
merge <- FindVariableFeatures(merge, selection.method = "vst", nfeatures = 2000)
  
#scale data and run PCA
merge <- ScaleData(merge, verbose = FALSE, vars.to.regress = c("total_counts","pct_counts_mito"))
merge <- RunPCA(merge, npcs = 30, verbose = FALSE)

merge <- merge %>% 
  RunHarmony("batch", plot_convergence = TRUE)
    
#batch-correct embedding
harmony_embeddings <- Embeddings(merge, 'harmony')
  
elbow<-ElbowPlot(merge, reduction = "harmony")
  
merge <- merge %>% 
    RunUMAP(reduction = "harmony", dims = 1:15) %>% 
    FindNeighbors(reduction = "harmony", dims = 1:15) %>% 
    FindClusters(resolution = 0.5) %>% 
    identity()
```

This works fine, and i want to confirm that it is ok to scale out other variable such as total counts and mitochondrial counts prior to harmony batch correction. I have seen one other post where this is carried out when scaling data so i believe this should be fine?

thanks for your time!

**Comment by slowkow** [comment]
**TL;DR** Try it! Prove to your audience that your chosen analysis gives the best result.

When solving a mathematics problem, we can often say with complete certainty that some answers are incorrect and some other answers are correct. We can even give our audience formal proofs to establish that the series of steps we used to arrive at an answer is valid and sound.

The situation is different in the context of data analysis and biological research. A vast number of questions can be asked about a particular dataset, but only some of those questions are of interest to our audience. Additionally, we may find that a single question can have multiple sensible answers, and it may not be immediately obvious which answer we prefer or why we prefer it. We may need to perform additional analyses and collect additional independent datasets to build confidence in our understanding. We might also need to reconsider the question.

In the context of this vast field of questions, we should aim to conduct an analysis with a limited set of specific questions in mind. Here are some examples of specific questions we might have for a single-cell RNA-seq dataset:
- Do we see a cell cluster of interest?
- What are the difference between cases and controls?

Before conducting our analysis, we should make sure that we have a specific question. This should help to focus our attention on the details that are relevant and avoid distractions. Discussion with advisors and peers can help to form the question and to understand if the question is sensible, worth asking, and answerable with the available data.

When conducting an analysis, there are many decisions to make along the way to the final result:
- Which observations should be included?
- Which features should be included?
- How should the data be normalized?
- Which model should be used?
- Which parameters should be used?

In the absence of a specific question that focuses our attention, we might get lost by trying to answer too many questions that are not of interest to our audience.

With our specific question in mind, we should ask:
- Does our chosen analysis produce valid or sensible results?
- Does a different analysis produce invalid or unexpected results?
- Does the choice at this step of the analysis fail to produce a noticeable difference on the final results?

Sometimes we want to "confirm that it is ok" to decide to use some particular step in an analysis. To do so, we should demonstrate the consequences of that step on the answer to the specific question that our audience cares about. When we report the results to our audience, we can justify our decision by showing the results with and without that step. We can try to show that a sensible result is not attainable if we choose a different option in the analysis, or that our choice gives the most desirable result.

If the choice at a particular step of the analysis does not seem to make a noticeable difference on the final results, then we might prefer the simpler analysis with fewer steps instead of the more complicated analysis with extra steps. This might help our audience to understand how we arrived at our conclusions.

After completing our analysis, we can attempt to validate our conclusions with an independent dataset. This might give our audience further justification for the entire series of choices we made in our analysis.
